Whitelisted variables (#436)
magento/merchdocs: src/marketing/email-template-message.md

title: Message Templates
---

> The process of customizing the body of each message is the same as the customizing the header or footer. The only difference is that there is a different message template for each activity or event that triggers a notification. You can use the templates as they are, or customize them to match your voice and brand. In addition to the template text, there is a wide selection of whitelisted [predefined]({% link marketing/variables-predefined.md %}) variables and [custom]({% link marketing/variables-custom.md %}) varibles that you can create and incorporate into the template.

Reviewer: ```suggestion
The process for customizing the body of each message is the same as the customizing the header or footer. The only difference is that there is a different message template for each activity or event that triggers a notification. You can use the templates as they are, or you can customize them to match your voice and brand. In addition to the template text, there is a wide selection of whitelisted [predefined]({% link marketing/variables-predefined.md %}) variables and [custom]({% link marketing/variables-custom.md %}) variables that you can create and incorporate into the template.
```

## Step 1. Load the default template

1. On the _Admin_ sidebar, go to **Marketing** > _Communications_ > **Email Templates**.

1. Click <span class="btn">Add New Template</span>.

   ![Email templates - load default template]({% link images/images/email-templates-message-load-default.png %}){: .zoom}
   _Template Information_

1. Do the following:

    - Under **Load default template**, choose the **Template** that you want to customize.

    - Click <span class="btn">Load Template</span>.

## Step 2. Customize the template

> 1. In the **Template Name** field, enter a name for your custom template.

Reviewer: ```suggestion
1. For **Template Name**, enter a name for your custom template.
```

1. If needed, change the **Template Subject**.

>    This is the first line of the message, which is the salutation by default. You can leave it as is, or enter something more descriptive.

Reviewer: ```suggestion
   This is the first line of the message, which is the salutation by default. You can leave it as is, or you can enter something more descriptive.
```

    {:.bs-callout-info}
    When working in the template code, be careful not to accidentally type over the code that is enclosed in double braces.

1. To insert a variable, position the cursor in the code where you want the variable to appear.

>    The selection of variables varies by template, and includes whitelisted predefined and custom variables, if avalable.

Reviewer: ```suggestion
   The selection of variables varies by template and includes whitelisted [predefined]({% link marketing/variables-predefined.md %}) and [custom]({% link marketing/variables-custom.md %}) variables, if available.
```

1. Click <span class="btn">Insert Variable</span> and choose the variable that you want to insert.

   ![Email templates - insert variable]({% link images/images/email-template-insert-variable.png %}){: .zoom}
   _Insert Variable_

   A command to insert the variable is enclosed in curly braces and added to the code at the cursor location. For example:

   `customVar code=my_custom_variable`

> 1. To make CSS declarations, enter the styles in the **Template Styles** box.

Reviewer: ```suggestion
1. To make CSS declarations, enter the styles in **Template Styles**.
```

## Step 3. Update the configuration

> The Currently Used For breadcrumb trail at the top of the Template Information section shows where the template is used. In this example the template configuration is located on the Customer Configuration page, in the Create New Account Options section, and in the Default Welcome Email field.

Reviewer: ```suggestion
The Currently Used For breadcrumb trail at the top of the Template Information section shows where the template is used. In this example, the template configuration is located on the Customer Configuration page, in the Create New Account Options section, and in the Default Welcome Email field.
```

|Page|Customer Configuration|
|Section|Create New Account Options|
|Field|Default Welcome Email|

1. In the **Currently Used For** breadcrumb trail, click the link to open the template configuration page.

    ![]({% link images/images/email-template-new-currently-used-for.png %}){: .zoom}
    _Current email template_

1. Expand ![Expansion selector]({% link images/images/btn-expand.png %}) the section, and find the field for the email template that you customized.

> 1. Clear the **Use system value** checkbox, and click the name of your custom template.

Reviewer: ```suggestion
1. Clear the **Use system value** checkbox and click the name of your custom template.
```
